scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,350 @@
+        }
+        if (args.containsKey("ftpPort")) {
+            ftpPort = Integer.parseInt(args.get("ftpPort").toString());
+        }
+        if (args.containsKey("ftpMode")) {
+            ftpMode = args.get("ftpMode").toString();
+        }
+        if (args.containsKey("ftpExtractArchive")) {
+            ftpExtractArchive = Boolean.parseBoolean(args.get("ftpExtractArchive").toString());
+        }
+        if (args.containsKey(FTP_LOCAL_FILE_PATH) && !args.get(FTP_LOCAL_FILE_PATH).toString().isEmpty()) {
+            ftpLocalFilePath = args.get(FTP_LOCAL_FILE_PATH).toString();
+        } else {
+            //We throw an exception tp prevent transferring all the contents of the global space.
+            if (ftpMode.equals("PUT")) {
+                throw new IllegalArgumentException("You have to specify the local relative path. Empty value is not allowed.");
+            }
+            //Default value is getLocalSpace() because it will always be writable and moreover can be used to transfer files to another data space (global, user)
+            ftpLocalFilePath = getLocalSpace();
+        }
+        if (args.containsKey("ftpRemoteFilePath")) {
+            ftpRemoteFilePath = args.get("ftpRemoteFilePath").toString();
+        }
+
+        ftpUsername = getThirdPartyCredential("FTP_USERNAME");
+        ftpPassword = getThirdPartyCredential("FTP_PASSWORD");
+        if (ftpUsername == null || ftpPassword == null) {
+            throw new IllegalArgumentException("You first need to add your ftp username and password (FTP_USERNAME, FTP_PASSWORD) to the third party credentials");
+        }
+    }
+
+    /**
+     * @see JavaExecutable#execute(TaskResult[])
+     */
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException {
+        List<String> filesRelativePathName;
+        FTPClient ftpClient = new FTPClient();
+
+        // connect and login to the server
+        ftpClient.connect(ftpHostname, ftpPort);
+        int reply = ftpClient.getReplyCode();
+        if (!FTPReply.isPositiveCompletion(reply)) {
+            ftpClient.disconnect();
+            throw new IOException("Exception in connecting to FTP Server");
+        }
+        ftpClient.login(ftpUsername, ftpPassword);
+
+        // use local passive mode to pass firewall
+        ftpClient.enterLocalPassiveMode();
+
+        getOut().println("Connected");
+
+        switch (ftpMode) {
+            //FTP mode is GET
+            case "GET":
+                filesRelativePathName = ftpGet(ftpClient);
+                break;
+
+            //FTP mode is PUT
+            case "PUT":
+                filesRelativePathName = ftpPut(ftpClient);
+                break;
+
+            default:
+                throw new IllegalArgumentException("FTP MODE can only be PUT or GET.");
+        }
+        // log out and disconnect from the server
+        ftpClient.logout();
+        ftpClient.disconnect();
```

Comment:
If we threw this IllegalArgumentException("FTP MODE can only be PUT or GET."), then will will not logout() and disconnect(). I do not know how severe is this. 
but in case you can:
```
try {
...
} finally {
  ftpClient.logout();
  ftpClient.disconnect();
}
```